### ISSUE TITLE
Support multi input and output

### ISSUE THREAD

**Issue by fernandocamargoti** [pytorchbearer/torchbearer#629]
Right now, it's not possible to:
- Have a Module multiple inputs (eg. forward(x1, x2)).
- Have a Module with multiple outputs (returning a tuple).

I worked around the first problem by creating a Module with a single input and indexing each individual input.
But the second problem makes it impossible to use the TripletMarginLoss, for example, since it expects 3 outputs from the Module.

**Comment by MattPainter01** [comment]
Hi, thanks for the issue!

I've had a quick look at the list of PyTorch losses and TripletMarginLoss seems to be the only one that expects input other than of the form (x, y). As such I'd rather not add a special case just for this. 

A simple work around is to wrap the triplet loss with a function that expects (y_pred, y_true) and return the evaluated triplet loss after taking the anchor, positive and negative values from the module output. Heres an example of this based off the PyTorch doc example: 

```python
from torchbearer import Trial
import torch
from torch import nn

triplet_loss = nn.TripletMarginLoss(margin=1.0, p=2)


class RandModule(nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, _):
        anchor = torch.randn(100, 128, requires_grad=True)
        positive = torch.randn(100, 128, requires_grad=True)
        negative = torch.randn(100, 128, requires_grad=True)
        return anchor, positive, negative


def crit(y_pred, _):
    return triplet_loss(*y_pred)


trial = Trial(RandModule(), criterion=crit).for_steps(100).run(1)
```

Better support for this would be nice though, do you have any suggestions on how you'd like to see this work?

**Comment by fernandocamargoti** [comment]
Hello, @MattPainter01. Thanks for your fast answer.

I'm preparing a pull request right now and I'll probably finish it in the next hour.

**Comment by fernandocamargoti** [comment]
@MattPainter01, could you review my pull request https://github.com/pytorchbearer/torchbearer/pull/630?

**Comment by fernandocamargoti** [comment]
It seems that I used a syntax that it's not compatible with Python 2.7. I'll try to fix it now,
